### PR TITLE
Move File Integration test

### PR DESCRIPTION
**Pull request by Tulsishah**
This will be pull request template
### Description

Adding Integration tests for moving file.
1. Within same directory
2. Within different directory

### Testing details
1. Manual - All the tests are passing
2. Unit tests - N/A
3. Integration tests - All the tests from integration tests directory working fine together.

### CODE REVIEW

File: tools/integration_tests/fileoperations/move_file_test.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+	}
+	defer file.Close()
+
+	_, err = file.WriteString(moveFileContent)
+	if err != nil {
+		t.Errorf("File at %v", err)
+	}
+}
+
+func createDestinationDirectoryAndMoveFile(t *testing.T, destDirPath string, srcFilePath string) {
```
Comment (raj-prince) [inline tools/integration_tests/fileoperations/move_file_test.go:60]: Let's stick with the same order of function parameter. Somewhere `testing.T` is in the first and somewhere it is in the last.

File: tools/integration_tests/fileoperations/fileoperations_test.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/setup"
+)
+
+const moveFile = "move.txt"
+const moveFileContent = "This is from move file in Test directory.\n"
+
+func TestMain(m *testing.M) {
+	setup.ParseSetUpFlags()
+
+	flags := [][]string{{"--implicit-dirs"}}
```
Comment (raj-prince) [inline tools/integration_tests/fileoperations/fileoperations_test.go:32]: We should also enable it for without --implicit-dirs.

File: tools/integration_tests/fileoperations/move_file_test.go [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+)
+
+// Move file from src directory to destination.
+func checkIfMovingFileSucceeded(srcDirPath string, destDirPath string, t *testing.T) {
+	cmd := exec.Command("mv", srcDirPath, destDirPath)
+
+	err := cmd.Run()
+	if err != nil {
+		t.Errorf("Moving file operation is failed.")
+	}
```
Comment (raj-prince) [inline tools/integration_tests/fileoperations/move_file_test.go:35]: Since, we have moving file and content, then we can also add check for the content.

**Review by raj-prince — COMMENTED** [review]
A few comments...

File: tools/integration_tests/fileoperations/move_file_test.go (hunk 3 above)
Comment (Tulsishah) [reply tools/integration_tests/fileoperations/move_file_test.go]: We already had.

File: tools/integration_tests/fileoperations/move_file_test.go (hunk 1 above)
Comment (Tulsishah) [reply tools/integration_tests/fileoperations/move_file_test.go]: fixed

**Comment by raj-prince** [comment]
Changes look fine.
Are we covering rename file case using the `mv` command, E.g. `mv a.txt b.txt`? It's same as moving within the same directory with different file-name.

**Comment by Tulsishah** [comment]
> ring rename file case using the

Yes it is 

> Changes look fine. Are we covering rename file case using the `mv` command, E.g. `mv a.txt b.txt`? It's same as moving within the same directory with different file-name.

Isn't it the case to rename the file - mentioned in this [test case](https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/tools/integration_tests/operations/file_operations_test.go#L28)

File: tools/integration_tests/operations/move_file_test.go [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+		t.Errorf("Error in creating file %v:", err)
+	}
+
+	err = operations.WriteFile(file.Name(), MoveFileContent)
+	if err != nil {
+		t.Errorf("File at %v", err)
+	}
+}
+
+func checkIfFileMovingSucceeded(srcFilePath string, destDirPath string, t *testing.T) {
```
Comment (raj-prince) [inline tools/integration_tests/operations/move_file_test.go:33]: checkIfFileMoveOperationSucceeded

File: tools/integration_tests/operations/move_file_test.go [hunk 5]
```diff
@@ -0,0 +1,86 @@
+package operations_test
```
Comment (raj-prince) [inline tools/integration_tests/operations/move_file_test.go:1]: Add the license header.

**Review by raj-prince — APPROVED** [review]
Minor comment - please incorporate before merging.

**Pull request merged**